packages/amplify-graphql-generation-transformer/src/validation.ts

```diff
@@ -0,0 +1,64 @@
+import { InvalidDirectiveError } from '@aws-amplify/graphql-transformer-core';
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { GenerationDirectiveConfiguration } from './grapqhl-generation-transformer';
+import { isQueryType } from '@aws-amplify/graphql-transformer-core/src/utils';
+
+/**
+ * Validates the configuration for the `@generation` directive.
+ *
+ * This function performs validation checks on the provided configuration
+ * to ensure it meets the requirements for the `@generation` directive.
+ *
+ * @param {GenerationDirectiveConfiguration} config - The configuration object for the `@generation` directive.
+ * @param {TransformerContextProvider} ctx - The transformer context provider.
+ * @throws {InvalidDirectiveError} If the configuration is invalid.
+ */
+
+export const validate = (config: GenerationDirectiveConfiguration, ctx: TransformerContextProvider): void => {
+  validateFieldType(config);
+  validateInferenceConfig(config);
+};
+
+/**
+ * Validates the field type for the `@generation` directive.
+ *
+ * This function checks if the parent of the field with the `@generation` directive
+ * is of type 'Query'. If not, it throws an InvalidDirectiveError.
+ *
+ * @param {GenerationDirectiveConfiguration} config - The configuration object for the `@generation` directive.
+ * @throws {InvalidDirectiveError} If the parent is not of type 'Query'.
+ */
+const validateFieldType = (config: GenerationDirectiveConfiguration): void => {
+  const { parent } = config;
+  if (!isQueryType(parent.name.value)) {
+    throw new InvalidDirectiveError('@generation directive must be used on Query field.');
+  }
+};
+
+/**
+ * Validates the inference configuration for the `@generation` directive according to the Bedrock API docs.
+ * {@link https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InferenceConfiguration.html}
+ * @param config The generation directive configuration to validate.
+ */
+const validateInferenceConfig = (config: GenerationDirectiveConfiguration): void => {
```

Comment:
As noted elsewhere, let's make sure we have unit tests that exercise these validations, especially at the boundaries.

Reply:
[Here they are](https://github.com/aws-amplify/amplify-category-api/pull/2820/files#diff-0d72d63c893ce32657c4a3ace88145f7c5d65da8a5e0f5b0c70d0303cc619d5fR144-R146).